### DISCUSSION TITLE
SSH Agent with KeepassXC portable on Windows

### DISCUSSION THREAD

**Discussion by DarrenSchwartz** [keepassxreboot/keepassxc#8091] [Q&A]
I have searched high and low including a careful reading of the user guide. I cannot get KeepassXC portable on Windows to work with Pagent or OpenSSH. In settings, it says "SSH Agent connection is Working", but the `SSH_AUTH_SOCK value` is empty and the `SSH_SK_PROVIDER value` is empty. I cannot find any guidance for what to put in the override field. I have not tried with the installed version, only portable, so I don't know if it is a portable issue.
Please help!

EDIT: Tried with installed version. Still not working.

**Comment by droidmonkey** [comment]
You followed these instructions exactly? https://keepassxc.org/docs/KeePassXC_UserGuide.html#_openssh_agent_and_pageant_on_windows

**Reply by DarrenSchwartz** [reply]
yes, exactly. There is this line:

> If the value of SSH_AUTH_SOCK is empty it means the agent is not properly configured and KeePassXC will be unable to connect to it unless you provide a static override path to the socket.

And that is exactly my problem. But I have no idea how to find the path to the socket. Note that KeepassXC does see the agent and shows a green banner at the top saying says 'SSH Agent connection is Working". Just the two values below say "empty".

**Reply by droidmonkey** [reply]
That only applies to Linux, not Windows. If KeePassXC is saying you are connected, then you can load keys into the agent.

**Reply by DarrenSchwartz** [reply]
When I try to load a key I get this banner:
```
Agent refused this identity. Possible reasons include: The key has already been added. A confirmation request is not supported by the agent (check options).
```

**Reply by droidmonkey** [reply]
Is it already added?? I just tested both Open SSH and Pageant on Windows and they both worked perfectly

**Reply by DarrenSchwartz** [reply]
No it wasn't already added, but I figured it out. I originally converted the key from dropbear to openssh format and was sure I used that one. I guess somethig was wrong with it because when I ran the convert again, it worked. Sorry for the goose chase and thanks for your help!
Maybe the error message could catch this type of wrong-format error and give some better direction?

**Reply by droidmonkey** [reply]
We are limited in what we can tell from the ssh agent... those programs don't return any useful messages or indicators.
